funnel/models/event.py

```diff
@@ -97,9 +132,38 @@ class Participant(BaseMixin, db.Model):
 
     __table_args__ = (db.UniqueConstraint('proposal_space_id', 'email'),)
 
+    @classmethod
+    def get(cls, space, email):
+        return cls.query.filter_by(proposal_space=space, email=email).one_or_none()
+
+    @classmethod
+    def upsert(cls, space, emailid, **fields):
+        participant = cls.get(space, emailid)
+        if participant:
+            participant._set_fields(fields)
+        else:
+            fields.pop('proposal_space', None)
+            fields.pop('email', None)
```

Comment:
Let's disambiguate by using a `current_` or `new_` prefix then. It's unclear why one is `space` and another `proposal_space` otherwise (or `emailid` and `email`).